ArticleTemplates/assets/js/modules/messenger.js

```diff
@@ -0,0 +1,205 @@
+/* global Promise */
+define([
+    'modules/post-message'
+], function (postMessage) {
+    'use strict';
+
+    var allowedHosts = [
+        location.protocol + '//' + location.host
+    ];
+    var listeners = {};
+    var registeredListeners = 0;
+
+    var error405 = { code: 405, message: 'Service %% not implemented' };
+    var error500 = { code: 500, message: 'Internal server error\n\n%%' };
+
+    return {
+        register: register,
+        unregister: unregister,
+        start: start
+    };
+
+    function start(modules) {
+        modules.forEach(function (module) {
+            module.init(register);
+        });
+
+        var iframes = document.getElementsByTagName('iframe');
+        var hostRX = /(https?):\/\/([\w\d-]+(?:\.[\w\d-]+)*(?:\:\d+))/
+        for(var i = 0, ii = iframes.length; i < ii; i++) {
+            if (!iframes[i].classList.contains('atom-embed')) {
+              continue;
+            }
+
+            var matches = iframes[i].src.match(hostRX);
+            if (!matches) {
+              continue;
+            }
+
+            allowedHosts.push(matches[1] + '://' + matches[2]);
+            iframes[i].contentWindow.postMessage('go', '*');
```

Comment:
Embedded iframes will be waiting for this message before sending anything. Now it is not 100% foolproof. Depending on the vagaries of network conditions, the embedded document may not be listening yet... I need to improve that part.